src/main/java/FRIDAY/Storage.java

```diff
@@ -0,0 +1,107 @@
+package FRIDAY;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * This method reads the storage file then loads the contents into an ArrayList of tasks, and finally returns the ArrayList
+     * @return ArrayList of Tasks
+     * @throws FRIDAYException exception is thrown when the file type in the storage file is not recognized
+     * @throws FileNotFoundException exception thrown when the storage file is unable to be found
+     */
+    public ArrayList<Task> load() throws FRIDAYException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File storageFile = new File(filePath);
+        try {
+            Scanner fileScanner = new Scanner(storageFile);
+            while (fileScanner.hasNextLine()) {
+                String data = fileScanner.nextLine();
+                populateProgramMemory(data, tasks);
+            }
+            fileScanner.close();
+        } catch(FileNotFoundException e) {
+            this.createDatabase();
+        }
+        finally {
+            return tasks;
+        }
+    }
+
+    private void createDatabase() throws FRIDAYException {
+        File db = new File(this.filePath);
+        File dir = new File(db.getParent());
+        dir.mkdir();
+        try {
+            db.createNewFile();
+        } catch (IOException e) {
+            throw new FRIDAYException("Error creating database");
+        }
+    }
+
+    /**
+     * method to populate the program storage with data read from the hard drive
+     *
+     * @param data string to represent all the details of the task, including its type and other miscellaneous details
```

Comment:
A good idea would be to make use of `@throws` tag in javadoc to signify the `FRIDAYException` being thrown.